### DISCUSSION TITLE
Is there any feature similar to OmitType and PickType in NestJS?

### DISCUSSION THREAD

**Discussion by cjnoname** [MichalLytek/type-graphql#921] [Q&A]
I want to pick or omit fields from an exiting model class as well as the decorators, which is similar to the function supported by NestJS named OmitType and PickType.

Please ignore my request if there's any similar funtions supported.

Thanks

**Comment by MichalLytek** [comment]
`OmitType` and `PickType` are just a workarounds that doesn't play well with other decorator-based libraries like TypeORM.

You should use ES6 class mixins pattern instead:
https://github.com/MichalLytek/type-graphql/tree/master/examples/mixin-classes

**Reply by cjnoname** [reply]
Thanks! Close now :)

**Reply by cjnoname** [reply]
Hi @MichalLytek, I had a deep think on it and I still believe that it's a kind of useful workaround.

Although the TypeORM decorator will not work on the derived class, we still can simply create the graphql classess with InputType or ObjectType from the entity class with all the types safe.

The Mixin is not quite handy to use and it can only extend but could not Omit or Pick

**Reply by MichalLytek** [reply]
Mixins allows you to compose the final classes from smaller parts. You can achieve the same results like with Omit or Pick.

**Reply by MichalLytek** [reply]
You can find on GH issues some code for those workarounds. They won't be officially available.
